Added enhancements on top of Calling Server preview 1

navali-msft opened the pull request:
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

zihzhan-msft committed:
Clean up clients comments.

navali-msft committed:
Models updates to introduce CallConnection and Server.

navali-msft committed:
callState updated to callConnectionState, replace sourceAlternateIdentity with alternateCallerId, other fixes

navali-msft committed:
Update to the test sample.

zihzhan-msft commented on sdk/communication/Azure.Communication.CallingServer/src/Generated/ServerCallRestClient.cs:
```diff
@@ -665,41 +437,41 @@ public Response StopRecording(string conversationId, string recordingId, Cancell
             uri.AppendPath("/recordings/", false);
             uri.AppendPath(recordingId, true);
-            uri.AppendPath("/Pause", false);
+            uri.AppendPath("/:pause", false);
```
is this colon add-on expected?

navali-msft replied:
Yes, part of azure api guideline.

zihzhan-msft commented on sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:
```diff
@@ -25,13 +25,13 @@ protected CommunicationIdentityClient CreateInstrumentedCommunicationIdentityCli
         /// </summary>
-        /// <returns>The instrumented <see cref="CallClient" />.</returns>
-        protected CallClient CreateInstrumentedCallingServerClient()
+        /// <returns>The instrumented <see cref="CallConnection" />.</returns>
```
I think the return should be CallingServerClient

navali-msft replied:
This is fixed.

navali-msft committed:
Updated the swagger path, removed delete operation, updated unit tests and live tests

navali-msft commented on sdk/communication/Azure.Communication.CallingServer/src/CallingServerClient.cs:
```diff
@@ -0,0 +1,295 @@
+    {
+        private readonly ClientDiagnostics _clientDiagnostics;
+        internal CallConnectionRestClient CallConnectionRestClient { get; }
+        internal ServerCallRestClient ServerCallRestClient { get; }
```
We are going with model where CallingServerClient will be creating CallConnection and ServerCall object. 

CallConnection can be created through CreateCallConnection or JoinCallConnection function and if you know the CallConnectionId then GetCallConnection can be used. 

ServerCall can be created by calling InitializeServerCall method.

CallConnection and ServerCall objects can then be used to perform different operations including PlayAudio, AddParticipant, RemoveParticipant and other.

navali-msft commented on sdk/communication/Azure.Communication.CallingServer/src/swagger.json:
```diff
@@ -0,0 +1,1521 @@
+{
```
Done

navali-msft committed:
Updates based on comments

ajaych-ms commented on sdk/communication/Azure.Communication.CallingServer/samples/Sample1_CreateCall.md:
```diff
@@ -4,13 +4,13 @@ This sample demonstrates how to make a call to a phone number recipient.
 To get started you'll need a Communication Service Resource.  See [README][README] for prerequisites and instructions.
 
-## Creating an `ServerCallingClient`
+## Creating an `CallingServerClient`
```
>an [](http://example.com/codeflow?start=12&length=2)

a

navali-msft replied:
fixed.

ajaych-ms replied:
I still see an?

minnieliu replied:
Still see "an" as well

zihzhan-msft replied:
Addressed now.

zihzhan-msft replied:
sms has similar grammar issue.

ajaych-ms replied:
still see an :)

zihzhan-msft replied:
No more "an"

ajaych-ms commented on sdk/communication/Azure.Communication.CallingServer/samples/Sample1_CreateCall.md:
```diff
@@ -27,12 +27,12 @@ var createCallOption = new CreateCallOptions(
     options: createCallOption // The options for creating a call.
     );
-Console.WriteLine($"Call Leg id: {createCallResponse.CallLegId}");
+Console.WriteLine($"Call connection id: {callConnection.Value.CallConnectionId}");
```
>.Value [](http://example.com/codeflow?start=55&length=6)

why do we need .Value?

navali-msft replied:
Apis are returning Response object which kind of wrapped http response object. The Value property contain the return value.

minnieliu replied:
I believe you can directly just get `callConnection.CallConnectionId` Can you give it a try?

ajaych-ms replied:
then variable should be named differently - something like callConnectionResponse

zihzhan-msft committed:
Merge branch 'feature/CallingServer-changes' into feature/callingServer-preview1-enhancement

zihzhan-msft committed:
Zihzhan/fix (#21727)

* Clean up clients comments.

* Update Snippet and Export-API.

zihzhan-msft committed:
Merge branch 'feature/callingServer-preview1-enhancement' of https://github.com/Azure/azure-sdk-for-net into feature/callingServer-preview1-enhancement

zihzhan-msft committed:
pull master without Downloader merge.

zihzhan-msft committed:
Zihzhan/fix (#21764)

* Update Snippet and Export-API.

* remove calloption override in servercall client.

* Nit for params and arguments handling.

* Merge Downloader updates.

* Merge Downloader updates test cases.

* Update AutoGen Content.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/src/Models/CallConnectionState.cs:
```diff
@@ -8,8 +8,8 @@ namespace Azure.Communication.CallingServer
-    [CodeGenModel("CallState", Usage = new string[] { "input", "output" }, Formats = new string[] { "json" })]
-    public readonly partial struct CallState
+    [CodeGenModel("CallConnectionState", Usage = new string[] { "input", "output" }, Formats = new string[] { "json" })]
+    public readonly partial struct CallConnectionState
```
Should we name this `CallConnectionStatus` instead of `CallConnectionState`. We have a call state in the calling client that seems to be different. This seems more of a status than a state.

navali-msft replied:
Status could be use if are referring to whether call is currently active or not, but state give its current stage in whole overall call process.

Calling client is using different verbs but meaning of some of them are same.

RezaJooyandeh replied:
> Calling client is using different verbs but meaning of some of them are same.

But should not we use the same terminology?

ajaych-ms replied:
Naveed changed the meaning to be the same as client SDK - connecting/connected/disconnecting/disconnected

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClients/CallingServerClientsTests.cs:
```diff
@@ -0,0 +1,122 @@
+
+        private static IEnumerable<object?[]> TestData_CreateCall()
+        {
+            return new List<object?[]>(){
```
I see this used around a lot. You can simplify replace most of the `new List<T>` code around with `new []`

navali-msft replied:
Done.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClients/TestBase.cs:
```diff
@@ -0,0 +1,63 @@
+
+namespace Azure.Communication.CallingServer.Tests
+{
+    public class TestBase
```
`CallingServerTestBase`

navali-msft replied:
Done.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClients/TestBase.cs:
```diff
@@ -0,0 +1,63 @@
+        {
+            var mockResponse = new MockResponse(responseCode);
+
+            if (responseContent != null)
```
C# let's you do stuff like:

```C#
if (responseContent is string responseContentString)
    mockResponse.SetContent(responseContentString);
```

or
```C#
switch (responseContent)
{
    case string responseContentString:
       mockResponse.SetContent(responseContentString);
    ....
}
```

navali-msft replied:
Done, thanks for pointing it out.

navali-msft commented on sdk/communication/Azure.Communication.CallingServer/src/CallConnection.cs:
```diff
@@ -15,217 +15,42 @@ namespace Azure.Communication.CallingServer
     /// The Azure Communication Services Calling Server client.
     /// </summary>
-    public class CallClient
+    public class CallConnection
```
Got okay from the board to use CallConnection and SeverCall during Java reviewing meeting.

zihzhan-msft commented on sdk/communication/Azure.Communication.CallingServer/src/Downloader/DownloadExtentions.cs:
```diff
@@ -0,0 +1,234 @@
+        /// A <see cref="RequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public static Response<Stream> DownloadStreaming(
```
I moved the download related method to this extension class. I 'd like to keep the CallingServerClient.cs clean and small.

ajaych-ms replied:
This has been fixed

navali-msft committed:
Model name updates (Suffix for responses models now has Result instead of Response), fixes based on comments (#21790)

navali-msft commented on sdk/communication/Azure.Communication.CallingServer/src/Generated/AzureCommunicationCallingServerServiceModelFactory.cs:
```diff
@@ -11,11 +11,11 @@ namespace Azure.Communication.CallingServer
     public static partial class AzureCommunicationCallingServerServiceModelFactory
```
This is fixed.

zihzhan-msft committed:
Zihzhan/fix (#21805)

* Update Snippet and Export-API.

* remove calloption override in servercall client.

* Nit for params and arguments handling.

* Merge Downloader updates.

* Merge Downloader updates test cases.

* Update AutoGen Content.

* Nit clean on Unit Tests.

* Updating and renaming.

navali-msft committed:
The changes which are done: (#21863)

InviteParticipants --> AddParticipant
InviteParticipantResultEvent --> AddParticipantResultEvent
CallModality --> MediaType
CommunicationParticipant --> CallParticipant
Id in PlayAudioResult & CancelAllMediaOperations is now called OperationId
CallConnectionState now has {incoming, connecting, connected, disconnecting, disconnected}
CallRecordingStateResult ---> CallRecording... by Naveed Ali
CallRecordingStateResult ---> CallRecordingProperties

zihzhan-msft committed:
Fix build Anaylse.

chrwhit committed:
fix(tests): improve code coverage

chrwhit committed:
Merge branch 'feature/callingServer-preview1-enhancement' of https://github.com/Azure/azure-sdk-for-net into feature/callingServer-preview1-enhancement

navali-msft committed:
Added AddParticipantResult, Re-added communication error, updated swagger path (#21870)

* Added AddParticipantResult, Re-added communication error, updated swagger path

* Fixed unit tests

* Updated netstandard

navali-msft committed:
Analysis fix

navali-msft committed:
Updated event type for add participant

zihzhan-msft committed:
Merge branch 'master' of https://github.com/Azure/azure-sdk-for-net into feature/callingServer-preview1-enhancement

zihzhan-msft committed:
Fix build issue.

zihzhan-msft committed:
Zihzhan/fix (#21905)

* Update Snippet and Export-API.

* remove calloption override in servercall client.

* Nit for params and arguments handling.

* Merge Downloader updates.

* Merge Downloader updates test cases.

* Update AutoGen Content.

* Nit clean on Unit Tests.

* Updating and renaming.

* Test update.

* Add Call Connection Live Tests.

* Add ServerCall Live Tests

* Increase Test Coverage.

* Reenable sample tests.

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/src/Models/Events/AddParticipantResultEvent.cs:
```diff
@@ -9,20 +9,20 @@ namespace Azure.Communication.CallingServer
     /// <summary>
     /// The invited participants result event.
```
Change invited to added

zihzhan-msft replied:
Resolved

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerTestBase.cs:
```diff
@@ -0,0 +1,61 @@
+{
+    public class CallingServerTestBase
+    {
+        protected const string dummyAccessKey = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9+eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ+SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV+adQssw5c=";
```
Where does this access key come from?

zihzhan-msft replied:
It's not been used anywhere, I removed it.

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/tests/CallingServerClient/CallingServerClientTests.cs:
```diff
@@ -0,0 +1,151 @@
+                        }
+                    )
+                    {
+                        AlternateCallerId = new PhoneNumberIdentifier("+17781234567"),
```
We usually try to use `+1425XXXXXX` as a fake number. `+17781234567` could actually be a real number

zihzhan-msft replied:
Changed to "+1425XXXXXXX".

minnieliu replied:
Hey Zihan, sorry I wasn't clear enough, I meant for `+1425XXXXXX` as a format, the X's should actually be numbers. The actual number we use as fake numbers is usually `+14255550123`

zihzhan-msft replied:
😃Addressed this issue. change it to +14255550123

zihzhan-msft commented on sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerLiveTestBase.cs:
```diff
@@ -46,9 +46,9 @@ protected async Task SleepIfNotInPlaybackModeAsync()
                 await Task.Delay(10000);
```
There are few Tests in Record mode required human interaction(i.e. pick up the call to let the call in established state, so that further Api operations can be performed). 
Alternative ways is using debug break point.

RezaJooyandeh replied:
👍. You can renaming the method then to `WaitToPickUpPhoneInPlaybackModeAsync` to make it more clear

zihzhan-msft committed:
Zihzhan/fix (#21931)

* Update Snippet and Export-API.

* remove calloption override in servercall client.

* Nit for params and arguments handling.

* Merge Downloader updates.

* Merge Downloader updates test cases.

* Update AutoGen Content.

* Nit clean on Unit Tests.

* Updating and renaming.

* Test update.

* Add Call Connection Live Tests.

* Add ServerCall Live Tests

* Increase Test Coverage.

* Reenable sample tests.

* Address comments.

* Point to swagger file in master branch.

zihzhan-msft committed:
Zihzhan/fix (#21950)

* Update Snippet and Export-API.

* remove calloption override in servercall client.

* Nit for params and arguments handling.

* Merge Downloader updates.

* Merge Downloader updates test cases.

* Update AutoGen Content.

* Nit clean on Unit Tests.

* Updating and renaming.

* Test update.

* Add Call Connection Live Tests.

* Add ServerCall Live Tests

* Increase Test Coverage.

* Reenable sample tests.

* Address comments.

* Point to swagger file in master branch.

* Nit fix.

* Nit changes based on Azure boarder review feedback.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/src/Models/CallingServerErrorResponse.cs:
```diff
@@ -0,0 +1,12 @@
+namespace Azure.Communication.CallingServer
+{
+    [CodeGenModel("CommunicationErrorResponse")]
+    public partial class CallingServerErrorResponse
```
Does this need to be public?

zihzhan-msft replied:
@navali-msft please take a look here.

navali-msft replied:
Removed CallingServerError as it is not used anywhere. Kept the one generated by sdk as internal.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/src/Models/CallingServerError.cs:
```diff
@@ -0,0 +1,12 @@
+namespace Azure.Communication.CallingServer
+{
+    [CodeGenModel("CommunicationError")]
+    public partial class CallingServerError
```
Does this need to be public?

zihzhan-msft replied:
@navali-msft

navali-msft replied:
Removed CallingServerError as it is not used anywhere. Keep the one generated by sdk as internal.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/src/Models/CallingServerModelFactory.cs:
```diff
@@ -0,0 +1,15 @@
+    /// Calling server model factory
+    /// </summary>
+    [CodeGenModel("AzureCommunicationServicesModelFactory")]
+    internal partial class CallingServerModelFactory
```
this probably should be moved to `.Models` namespace and also should it be public so developers can instantiate the models that do not have public constructor.
CC: @minnieliu

minnieliu replied:
Yup, agreed, good catch Reza!

ajaych-ms replied:
Is this a pattern followed by other modalities? So, far we don't seem to have this need for apps

RezaJooyandeh replied:
Yes. See this for more details: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/src/Azure.Communication.CallingServer.csproj:
```diff
@@ -1,13 +1,13 @@
   <PropertyGroup>
     <Description>
-      This client library enables working with the Microsoft Azure Communication Service Calling Server APIs.
+      This client library enables working with the Microsoft Azure Communication CallingServer service.
```
Can we rename to Calling Server, two separate words?

zihzhan-msft replied:
We prefer using "CallingServer" which was "Calling Server" to keep it consistent within SDK library and API

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/src/CallingServerClientOptions.cs:
```diff
@@ -39,7 +39,7 @@ public enum ServiceVersion
             /// </summary>
 #pragma warning disable CA1707 // Identifiers should not contain underscores
-            V2021_04_15_Preview1 = 0
+            V2021_06_15_Preview = 1
```
why is the enum starting with 1 but the 2021_04_15 version is removed?

zihzhan-msft replied:
This number started from 1 was per Azure boarder team suggest, they seems reserve 0 for other usage. CC @navali-msft 

"2021_04_15" was a "private review" version. There are plenty of major schema changes on the swagger and since we are not public preview yet, we decide to have "V2021_06_15_Preview" as our first public version.

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/src/Models/CallParticipant.cs:
```diff
@@ -0,0 +1,34 @@
+        /// <param name="identifier"> The communication identifier. </param>
+        /// <param name="participantId"> Participant Id. </param>
+        /// <param name="isMuted"> Is participant muted. </param>
+        public CallParticipant(CommunicationIdentifier identifier, string participantId, bool? isMuted)
```
should isMuted be nullable?

zihzhan-msft replied:
@ajaych-ms @navali-msft

navali-msft replied:
Fixed this.

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerTestEnvironment.cs:
```diff
@@ -15,13 +15,18 @@ public class CallingServerTestEnvironment : CommunicationTestEnvironment
+        public string SourcePhoneNumber => GetRecordedVariable(AlternateCallerId, options => options.IsSecret());
 
         /// <summary>
         /// The phone number associated with the source.
```
Should this be The phone number associated with the target.?

zihzhan-msft replied:
Good catch!         
Change it to:
/// The phone number required to make a pstn call.

minnieliu commented on sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerTestEnvironment.cs:
```diff
@@ -15,13 +15,18 @@ public class CallingServerTestEnvironment : CommunicationTestEnvironment
         /// The phone number associated with the source.
         /// </summary>
-        public string SourcePhoneNumber => GetRecordedVariable(AzurePhoneNumber, options => options.IsSecret());
+        public string SourcePhoneNumber => GetRecordedVariable(AlternateCallerId, options => options.IsSecret());
```
If we are going to change it to alternateCallerId, can we change the variable name as well?

zihzhan-msft replied:
We prefer to using the SourcePhoneNumber in comparasion with TargetPhoneNumber

minnieliu replied:
Can you clarify why?

zihzhan-msft replied:
We have target phone number, so here we have source phone number, considering later on, the source phone number could be a pstn number that is not from ACS resource, then we assign another number and differentiate it with the AlternateCallerId from ACS.

zihzhan-msft committed:
Address comments.

zihzhan-msft committed:
Naming changes.

zihzhan-msft commented on sdk/communication/Azure.Communication.CallingServer/tests/Infrastructure/CallingServerTestEnvironment.cs:
```diff
@@ -12,10 +12,12 @@ namespace Azure.Communication.CallingServer.Tests
     /// </summary>
     public class CallingServerTestEnvironment : CommunicationTestEnvironment
     {
+        public const string AlternateCallerId = "ALTERNATE_CALLERID";
```
@minnieliu @RezaJooyandeh 
Should we move this to [CommunicationTestEnvironment.cs](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/communication/Azure.Communication.Common/tests/CommunicationTestEnvironment.cs) and create a Variable in the ACS Live Test KV?

minnieliu replied:
We usually only add it to CommunicationTestEnvironment for variables that will be shared between all the Communication SDKs. I think it is fine to keep it in CallingServerTestEnvironment

JianpingChen commented on sdk/communication/Azure.Communication.CallingServer/src/Models/CallingServerErrorResponse.cs:
```diff
@@ -0,0 +1,12 @@
+namespace Azure.Communication.CallingServer
+{
+    [CodeGenModel("CommunicationErrorResponse")]
+    public partial class CallingServerErrorResponse
```
This class is not currently used, correct? If you plan to convert low level exceptions to this, you should only add this class when the conversion code is added. Otherwise, it is confusing

ajaych-ms replied:
@navali-msft could you please remove

navali-msft replied:
Removed the public version. Keep the one generated by sdk.

zihzhan-msft committed:
Change to static connection string.

zihzhan-msft committed:
Add methods that support Live Test.

zihzhan-msft committed:
Moved Recorded value inside the test method.

navali-msft committed:
Updated models definition based on new swagger plus updated events properties to readonly (#21980)

* Updated models definition based on new swagger, also fixed events to make properties readonly

* Update accessibility of internal models

* Update the file path in autorest

zihzhan-msft committed:
Update auto gen apis.

zihzhan-msft committed:
Merge branch 'main' of https://github.com/Azure/azure-sdk-for-net into feature/callingServer-preview1-enhancement

navali-msft committed:
Rewrite the event tests and fixed json for other tests (#21983)

zihzhan-msft committed:
Disable AddParticipant test in Live mode.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/tests.yml:
```diff
@@ -18,4 +18,6 @@ extends:
       # SKIP_PHONENUMBER_LIVE_TESTS skips certain phone number tests such as purchase and release
-      SKIP_PHONENUMBER_LIVE_TESTS: TRUE
\ No newline at end of file
+      SKIP_PHONENUMBER_LIVE_TESTS: TRUE
```
why is `SKIP_PHONENUMBER_LIVE_TESTS` here?

zihzhan-msft replied:
I saw this flag is widely used in all the other communication SDK projects, even this flag is not used in them.

RezaJooyandeh commented on sdk/communication/Azure.Communication.CallingServer/src/Models/CallingServerModelFactory.cs:
```diff
@@ -0,0 +1,15 @@
+
+using Azure.Core;
+
+namespace Azure.Communication.CallingServer
```
`.Models`

zihzhan-msft replied:
This is resolved.

RezaJooyandeh reviewed (approved):
I left another couple of small comments, but feel free to merge once you address those and @minnieliu approve as well.

navali-msft committed:
Moved modelfactory to .models namespace (#22021)
